### PR TITLE
Burgers 

### PR DESCRIPTION
**Pull request by SCiarella**
This PR introduces examples relative to the Burgers equation:
* [x] 03.01 2d B equation as CNODE  (generic DNS)
* [ ] 03.02  Learning a closure 
* [ ] 03.03  extra small-scale as additional equation
* [ ] ?

### CODE REVIEW

File: src/loss_posteriori.jl [hunk 1]
```diff
@@ -254,9 +173,10 @@ function loss_MulDtO_oneset(trajectory,
     list_starts = cat([trajectory[:, :, i] for i in starting_points]...,
         dims = 2)
     # Use the differentiable solver to get the predictions
-    pred, list_tr = predict_u_CNODE(list_starts, θ, st, training_CNODE, list_tr)
+    pred, target = predict_u_CNODE(list_starts, θ, st, nunroll, training_CNODE, list_tr)
     # the loss is the sum of the differences between the real trajectory and the predicted one
-    loss = sum(abs2, list_tr .- pred) ./ sum(abs2, list_tr)
+    loss = sum(abs2, target[:, :, 2:nunroll] .- pred[:, :, 2:nunroll]) ./
```
Comment (luisaforozco) [inline src/loss_posteriori.jl:178]: @SCiarella I am having some problems with the dimensions of vectors when calculating the loss, I do not understand why are you getting rid of the first time step? Is this considering that there is a fix overlap of 1?

File: src/FNO.jl [hunk 2]
```diff
@@ -33,66 +38,82 @@ struct FourierLayer{A, F} <: Lux.AbstractExplicitLayer
...
+function ((; dim_to_fft, Nxyz, kmax, cout, cin, σ)::FourierLayer)(x, params, state)
     ## Destructure params
-    ## The real and imaginary parts of R are stored in two separate channels
     W = params.spatial_weight
-    W = reshape(W, 1, 1, cout, cin)
+    W = reshape(W, state.placeholder..., cout, cin)
     R = params.spectral_weights
-    R = selectdim(R, 5, 1) .+ im .* selectdim(R, 5, 2)
+    ## The real and imaginary parts of R are stored in two separate channels
+    R = selectdim(R, state.last_dim, 1) .+ im .* selectdim(R, state.last_dim, 2)
```
Comment (luisaforozco) [inline src/FNO.jl:94]: Is `im` a typo here?

File: src/loss_posteriori.jl (hunk 1 above)
Comment (SCiarella) [reply src/loss_posteriori.jl:178]: the first step is the initial condition for the NODE, so by construction `target[:, :, 1] == pred[:, :, 1]`
